### PR TITLE
fix broken uwp migrator

### PR DESCRIPTION
**Pull request by rohit21agrawal**
uwp migrator was broken because of some recent lsl related changes that were introduced. this changeset fixes that issue and adds end-to-end test coverage for the scenario

Fixes: devdiv internal bug 454369

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/VsCoreProjectSystemServices.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ internal class VsCoreProjectSystemServices
 
         public IProjectSystemReferencesService References => throw new NotSupportedException();
 
-        public IProjectSystemService ProjectSystem => throw new NotSupportedException();
+        public IProjectSystemService ProjectSystem { get; private set; }
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectServices/VsCoreProjectSystemServices.cs]: just `get` it's a read-only attribute

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsCoreProjectSystem.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using System.Threading.Tasks;
+using Microsoft;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    class VsCoreProjectSystem : IProjectSystemService
+    {
+        public IVsProjectAdapter VsProjectAdapter { get; private set; }
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsCoreProjectSystem.cs]: the same

File: test/EndToEnd/tests/ServicesTest.ps1 [hunk 3]
```diff
@@ -14,38 +14,38 @@
...
-#     Assert-AreEqual $packageRefs[0].GetMetadataValue("Version") '5.2.2'
-# }
+function Test-MigrateVanilaUwpProjectJsonToPackageReference {
+    param(
+        $context
+    )
+
+    # Arrange
+    $p = New-UwpClassLibrary UwpClassLibrary1
+    $cm = Get-VsComponentModel
```
Comment (alpaix) [inline test/EndToEnd/tests/ServicesTest.ps1]: not used

File: test/EndToEnd/tests/ServicesTest.ps1 [hunk 4]
```diff
@@ -14,38 +14,38 @@
...
+    # Assert
+
+    # Check if runtimes were migrated correctly
+    $expectRuntimeIds = 'win10-arm;win10-arm-aot;win10-x86;win10-x86-aot;win10-x64;win10-x64-aot'
+    Assert-True($result.IsSuccess)
+    $actualRuntimes = Get-MsBuildPropertyValue $p 'RuntimeIdentifiers'
+    Assert-AreEqual $expectRuntimeIds $actualRuntimes
+
+    # Check if project.json file was deleted
+    Assert-True !(Test-Path (Join-Path $projectDir project.json))
```
Comment (alpaix) [inline test/EndToEnd/tests/ServicesTest.ps1]: `Assert-False`

File: test/EndToEnd/tests/ServicesTest.ps1 [hunk 5]
```diff
@@ -14,38 +14,38 @@
...
-#     Assert-True (Test-Path (Join-Path $projectDir (Join-Path Backup UwpClassLibrary1.csproj)))
-
-#     # Check if package reference was added correctly
-#     $packageRefs = @(Get-MsBuildItems $p 'PackageReference')
-#     Assert-AreEqual 1 $packageRefs.Count
-#     Assert-AreEqual $packageRefs[0].GetMetadataValue("Identity") 'Microsoft.NETCore.UniversalWindowsPlatform'
-#     Assert-AreEqual $packageRefs[0].GetMetadataValue("Version") '5.2.2'
-# }
+function Test-MigrateVanilaUwpProjectJsonToPackageReference {
+    param(
```
Comment (alpaix) [inline test/EndToEnd/tests/ServicesTest.ps1]: `[SkipTestForVS14()]`

File: test/TestExtensions/API.Test/InternalAPITestHook.cs [hunk 6]
```diff
@@ -152,5 +152,15 @@ public static bool BatchEventsApi(string id, string version)
                            batchStartIds[0].Equals(batchEndIds[0], StringComparison.Ordinal);
                 });
         }
+
+        public static IVsProjectJsonToPackageReferenceMigrateResult MigrateJsonProject(string projectName)
+        {
+            var migrator = ServiceLocator.GetComponent<IVsProjectJsonToPackageReferenceMigrator>();
+            return ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```
Comment (alpaix) [inline test/TestExtensions/API.Test/InternalAPITestHook.cs]: Does it need to be on UI thread?

File: NuGet.sln [hunk 7]
```diff
@@ -556,10 +556,13 @@ Global
...
 		{96255044-3776-439B-8496-C460DB9C4F97} = {A8F45E94-4CFD-478C-9FE2-8E6DEE7E0770}
 		{79266117-FEDD-45B3-B603-33A4B6E9F12F} = {D1549653-9631-4E16-9967-55427174A0DC}
+		{F4CA95DA-0045-4AC4-ABA0-560008CEB963} = {23CEFC88-9365-4464-A517-700224ECA033}
 		{34750BB3-4A21-433A-9C55-C051CA70C4AE} = {BB63CACA-866F-454F-BA4C-78B788D032BB}
 		{08F523EC-3C2A-4A00-A54C-2E54C5AC856B} = {BD1946CE-5544-4F28-A04A-9C3D51113E1A}
 		{506AF844-92E0-4404-BBFC-0AB073890A72} = {BD1946CE-5544-4F28-A04A-9C3D51113E1A}
 		{F813268A-1EA1-41E6-A42C-01E7F937E257} = {7323F93B-D141-4001-BB9E-7AF14031143E}
-		{F4CA95DA-0045-4AC4-ABA0-560008CEB963} = {23CEFC88-9365-4464-A517-700224ECA033}
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
```
Comment (alpaix) [inline NuGet.sln:565]: What does this do?

**Review by alpaix — CHANGES_REQUESTED** [review]
Few minor notes

File: test/TestExtensions/API.Test/InternalAPITestHook.cs (hunk 6 above)
Comment (rohit21agrawal) [reply test/TestExtensions/API.Test/InternalAPITestHook.cs]: yes

File: NuGet.sln (hunk 7 above)
Comment (rohit21agrawal) [reply NuGet.sln:565]: no idea, it was autogenerated

**Comment by rohit21agrawal** [comment]
@alpaix changes made per feedback

File: test/TestExtensions/API.Test/InternalAPITestHook.cs (hunk 6 above)
Comment (alpaix) [reply test/TestExtensions/API.Test/InternalAPITestHook.cs]: Do we require XAML team to switch before calling us?

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsCoreProjectSystem.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+using System.Threading.Tasks;
+using Microsoft;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    class VsCoreProjectSystem : IProjectSystemService
+    {
+        public IVsProjectAdapter VsProjectAdapter { get; }
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsCoreProjectSystem.cs]: `private`

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsCoreProjectSystem.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    class VsCoreProjectSystem : IProjectSystemService
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsCoreProjectSystem.cs]: `internal`

**Review by alpaix — APPROVED** [review]
Two more comments, then :shipit:

File: test/TestExtensions/API.Test/InternalAPITestHook.cs (hunk 6 above)
Comment (rohit21agrawal) [reply test/TestExtensions/API.Test/InternalAPITestHook.cs]: no you are right, the switching is done by the api itself so we don't need to do it here. have fixed it.

**Comment by rohit21agrawal** [comment]
@alpaix changes made

**Review by alpaix — APPROVED** [review]
:sparkles: LGTM

**Pull request closed without merging**